sensing/imu.cpp

```diff
@@ -50,169 +51,327 @@ extern "C"
+	// 		config_.accelerometer.mean_values[i] = 0.5 * ( config_.accelerometer.mean_values[i] + oriented_acc_[i] );
+	// 	}
+	// }
+
+	// // Do accelero scale calibration
+	// if( do_accelerometer_scale_calibration_ )
+	// {
+	// 	;
+	// }
+
+	// // Do gyroscope bias calibration
+	// if( do_gyroscope_bias_calibration_ )
+	// {
+	// 	for (i=0; i<3; i++)
+	// 	{
+	// 		config_.gyroscope.max_values[i]  = maths_f_max(config_.gyroscope.max_values[i], oriented_gyro_[i]);
+	// 		config_.gyroscope.min_values[i]  = maths_f_min(config_.gyroscope.min_values[i], oriented_gyro_[i]);
+	// 		config_.gyroscope.mean_values[i] = 0.5 * ( config_.gyroscope.mean_values[i] + oriented_acc_[i] );
+	// 	}
+	// }
+
+	// // Do gyroscope scale calibration
+	// if( do_gyroscope_scale_calibration_ )
+	// {
+	// 	;
+	// }
+
+	// // Do magnetometer bias calibration
+	// if( do_magnetometer_bias_calibration_ )
+	// {
+	// 	for (i=0; i<3; i++)
+	// 	{
+	// 		config_.magnetometer.max_values[i]  = maths_f_max(config_.magnetometer.max_values[i], oriented_mag_[i]);
+	// 		config_.magnetometer.min_values[i]  = maths_f_min(config_.magnetometer.min_values[i], oriented_mag_[i]);
+	// 		config_.magnetometer.mean_values[i] = 0.5 * ( config_.magnetometer.mean_values[i] + oriented_acc_[i] );
+	// 	}
+	// }
+
+	// // Do magnetometer scale calibration
+	// if( do_magnetometer_scale_calibration_ )
+	// {
+	// 	;
+	// }
+
+	// Scale sensor values
+	for( int8_t i = 0; i < 3; i++ )
 	{
-		for (i=0; i<3; i++)
-		{
-			imu->calib_compass.max_oriented_values[i] = maths_f_max(imu->calib_compass.max_oriented_values[i],imu->oriented_magneto.data[i]);
-			imu->calib_compass.min_oriented_values[i] = maths_f_min(imu->calib_compass.min_oriented_values[i],imu->oriented_magneto.data[i]);
-		}
+		// 1: Unbias, 
+		// 2: scale 
+	// 	scaled_acc_[i]  = (1.0f - config_.lpf_acc ) * scaled_acc_[i]  + config_.lpf_acc  * ( ( oriented_acc_[i]  - config_.accelerometer.bias[i] ) * config_.accelerometer.scale_factor[i]  );
+	// 	scaled_gyro_[i] = (1.0f - config_.lpf_gyro) * scaled_gyro_[i] + config_.lpf_gyro * ( ( oriented_gyro_[i] - config_.gyroscope.bias[i]     ) * config_.gyroscope.scale_factor[i]     		);
+	// 	scaled_mag_[i] 	= (1.0f - config_.lpf_mag ) * scaled_mag_[i]  + config_.lpf_mag  * ( ( oriented_mag_[i]  - config_.magnetometer.bias[i]  ) * config_.magnetometer.scale_factor[i]  	);
+
+		// 1: scale
+		// 2: unbias
+		scaled_acc_[i]  = config_.lpf_acc  	* (oriented_acc_[i]   * config_.accelerometer.scale_factor[i] - config_.accelerometer.bias[i])
+				+ (1.0f - config_.lpf_acc ) * scaled_acc_[i];
+		scaled_gyro_[i] = config_.lpf_gyro  * ( oriented_gyro_[i] * config_.gyroscope.scale_factor[i]    - config_.gyroscope.bias[i]     ) 
+				+ (1.0f - config_.lpf_gyro) * scaled_gyro_[i];
+		scaled_mag_[i] 	= config_.lpf_mag   * ( oriented_mag_[i]  * config_.magnetometer.scale_factor[i] - config_.magnetometer.bias[i]  ) 
+				+ (1.0f - config_.lpf_mag ) * scaled_mag_[i];
```

Comment:
To discuss @GregoireH @ndousse @basil-huber :
I made this change because the scales are never updated, and it makes it easier to update the biases. 
The drawback is we need to update all calibrations by multiplying the biases by the scale factor. 
It can be reverted quickly in case you do not agree.